node-data-tools/index.js

```diff
@@ -0,0 +1,247 @@
+      csvStream.write(CSV.stringify(TidepoolDataTools.allFields));
+      processingStream
+        .pipe(es.mapSync(
+          data => CSV.stringify(TidepoolDataTools.allFields.map(field => data[field] || '')),
+        ))
+        .pipe(csvStream);
+    }
+
+    // Multiple CSVs
+    if (_.includes(command.outputFormat, 'csvs') || _.includes(command.outputFormat, 'all')) {
+      if (!fs.existsSync(outFilename)) {
+        fs.mkdirSync(outFilename);
+      }
+      Object.keys(config).forEach((key) => {
+        const csvStream2 = fs.createWriteStream(`${outFilename}/${key}.csv`);
+        csvStream2.write(CSV.stringify(Object.keys(config[key])));
+        events.EventEmitter.defaultMaxListeners += 2;
+        processingStream
+          // eslint-disable-next-line consistent-return
+          .pipe(es.mapSync((data) => {
+            if (data.type === key) {
+              return CSV.stringify(Object.keys(config[key]).map(field => data[field] || ''));
+            }
+          }))
+          .pipe(csvStream2);
+      });
+    }
+
+    // XLSX
+    if (_.includes(command.outputFormat, 'xlsx') || _.includes(command.outputFormat, 'all')) {
+      const xlsxStream = fs.createWriteStream(`${outFilename}.xlsx`);
+
+      events.EventEmitter.defaultMaxListeners += 1;
+      processingStream
+        .pipe(TidepoolDataTools.xlsxStreamWriter(xlsxStream));
+    }
+
+    readStream.on('end', () => {
+      console.info(`Exported ${counter} records.`);
+    });
+  });
+}
+
+/*
+function getData() {
+  // Implement this command
+}
+*/
+
+if (require.main === module) {
+  program
+    .name('tidepool-data-tools')
+    .version('0.1.0');
+
+  let commandInvoked = false;
+
+  program
+    .command('convert')
+    .description('Convert data between different formats')
+    .option('-i, --input-tidepool-data <file>', 'csv, xlsx, or json file that contains Tidepool data')
+    .option('-c, --config <file>', 'a JSON file that contains the field export configuration')
+    .option('--salt <salt>', 'salt used in the hashing algorithm', 'no salt specified')
+    .option('-o, --output-data-path <path>', 'the path where the data is exported',
+      path.join(__dirname, 'example-data', 'export'))
+    .option('-f, --output-format <format>', 'the path where the data is exported', (val, list) => {
+      list.push(val);
+      return list;
+    }, [])
+    // TODO: Implement options below this TODO
+    .option('--start-date [date]', 'filter data by startDate and EndDate')
+    .option('--end-date [date]', 'filter data by startDate and EndDate')
+    .option('--merge-wizard-data', 'option to merge wizard data with bolus data. Default is true')
+    .option(
+      '--filterByDatesExceptUploadsAndSettings',
+      'upload and settings data can occur before and after start and end dates, so include ALL upload and settings data in export',
+    )
+    .action((command, options) => {
+      convert(command, options);
+      commandInvoked = true;
+    });
+
+  /*
```

Comment:
I think it would be great to include an example of how to use getdata with this tool in the readme file

Reply:
There currently _is_ no getdata ;)

Reply:
let's chat about this, I must have misunderstood something you said a few weeks ago.